Add option to not include the Data API initialization code

marcelpanse opened the pull request:
We use bootstrap native to build a component. This component can be used on our clients' websites by including our javascript file. As soon as they include our library, bootstrap-native will start searching the entire DOM tree to find data- elements and start initializing them. This will have unintended side-effects. We only want to initialize our own inserted DOM elements programmatically, so I added an option to exclude the utils-init.js code. 
This solved #180

thednp commented:
We would also need this stuff for V3 as well, also some documentation update should also be taken into account.

marcelpanse commented:
If you agree on the implementation like this and want to potentially merge
it, then I'd happily take a look at v3 and the docs..?

Op vr 11 mei 2018 19:30 schreef thednp <notifications@github.com>:

> We would also need this stuff for V3 as well, also some documentation
> update should also be taken into account.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/thednp/bootstrap.native/pull/214#issuecomment-388431553>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAZkqI3rEfMDghqwyq_kbYftFRvg9bEEks5txcqugaJpZM4T7d_p>
> .
>

thednp commented:
I would rename the option to `autoInitDataAPI` and set this option to `true` by default.

marcelpanse commented:
I changed the option to `autoInitDataAPI` and made it `true` by default. 
I added it to the v3 version as well. 

Lastly, I updated the wiki page with the new option but reverted it again since this PR is not merged yet (you can see it in the history what I wanted to change). 

Let me know if there is anything else you need to get this merged.

marcelpanse commented:
@thednp Awesome, thanks! Could you build and publish a new version to NPM so I can use it?

midzer commented:
@marcelpanse you can use latest master in your `package.json` by putting `thednp/bootstrap.native` in your bootstrap.native version field.

thednp commented:
I added a note on the [wiki page](https://github.com/thednp/bootstrap.native/wiki/How-to-use), please confirm it's validity. Thanks

marcelpanse commented:
Looks good :-)

Op za 12 mei 2018 10:00 schreef thednp <notifications@github.com>:

> I added a note on the wiki page
> <https://github.com/thednp/bootstrap.native/wiki/How-to-use>, please
> confirm it's validity. Thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/thednp/bootstrap.native/pull/214#issuecomment-388538363>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAZkqCiWd_1xf1jklql1PUTm02noxwXEks5txpa0gaJpZM4T7d_p>
> .
>

thednp commented:
One more thing: what would be the most common use cases for this implementation? I'm thinking about mentioning those in the wikies as well.

marcelpanse commented:
If you are making a component that can be included on any page (like a widget/bar/popup/carousel/etc) then you'd want that component to be as small as possible but still have the benefits of using the Bootstrap components, so you use bootstrap.native.
Your component runs on a 3rd party website which you have no control over and might already use Bootstrap. To not influence the surrounding website you should set the `autoInitDataAPI` to `false` to make sure not to trigger other data- elements on the page by accident.
